### PR TITLE
Add KOPS_ARGS for passing arbitrary args to 'kops create cluster'

### PR DESCRIPTION
**Pull request by wongma7**
This small change could let me pass any arbitrary args without having to continue making too many PRs/changes to kops.go (or worse somehow maintain my own fork of it.) 

I agree for sure that a 'kops cli pass-thru' as mentioned here https://github.com/kubernetes/test-infra/pull/2019 would be ideal but since I'm apparnetly the only one using these scripts outside of official CI and I'm not sure how to implement it properly, this should be enough for now :)

### CODE REVIEW

**Comment by justinsb** [comment]
Well I for one think this would be handy, but not sure how this fits with the general test-infra plans :-)

File: kubetest/kops.go [hunk 1]
```diff
@@ -137,6 +137,9 @@ func (k kops) Up() error {
 	if k.image != "" {
 		createArgs = append(createArgs, "--image", k.image)
 	}
+	if os.Getenv("KOPS_ARGS") != "" {
```
Comment (fejta) [inline kubetest/kops.go]: Please make this a kubetest flag. `--kopsArgs` `--kops-args` or something similar.

**Comment by fejta** [comment]
Sounds super! Please just make it a flag that we pass into kubetest rather than a magical environment variable

File: kubetest/kops.go (hunk 1 above)
Comment (wongma7) [inline kubetest/kops.go]: Ok, I'm now working this now but having a lot of trouble plumbing --kops-args through the python scenario script->e2e-runner.sh->kubetest. Can you see anything obvious I'm doing wrong in the new PR code, any help would be appreciated!

printenv shows that E2E_OPT looks fine coming out of the python script and into e2e-runner.sh, kops-args value is quoted because there are spaces:
```
E2E_OPT=--deployment kops ... --kops-args="--dns private" ...
```
but later after various e2e_go_args are added, kubetest is ultimately called like this:
```
+(/workspace/e2e-runner.sh:179): main(): kubetest --deployment kops... '--kops-args="--dns' 'private"' ...
```

So I think there is some bash stuff happening that I don't quite understand how to get around.

File: kubetest/kops.go [hunk 2]
```diff
@@ -137,6 +140,10 @@ func (k kops) Up() error {
 	if k.image != "" {
 		createArgs = append(createArgs, "--image", k.image)
 	}
+	if k.args != "" {
+		createArgs = append(createArgs, strings.Split(os.Getenv(k.args), " ")...)
+	}
+	return fmt.Errorf("createArgs %s", createArgs)
```
Comment (fejta) [inline kubetest/kops.go]: Is this for debugging?

File: kubetest/main.go [hunk 3]
```diff
@@ -185,6 +185,8 @@ func main() {
 	log.SetFlags(log.LstdFlags | log.Lshortfile)
 	o := defineFlags()
 	flag.Parse()
+	log.Printf("%s", *kopsArgs)
```
Comment (fejta) [inline kubetest/main.go]: Please remove

**Comment by wongma7** [comment]
@fejta sorry, I've now pushed the cleaned up & working patch.

I was having trouble getting the args through bash but the change to e2e-runner.sh should fix it.

File: jenkins/e2e-image/e2e-runner.sh [hunk 4]
```diff
@@ -176,4 +176,5 @@ if [[ "${E2E_PUBLISH_GREEN_VERSION:-}" == "true" ]]; then
   e2e_go_args+=(--publish="gs://${KUBE_GCS_DEV_RELEASE_BUCKET}/ci/latest-green.txt")
 fi
 
-kubetest ${E2E_OPT:-} "${e2e_go_args[@]}" "${@}"
+declare -a 'e2e_opt_args=('"$E2E_OPT"')'
```
Comment (fejta) [inline jenkins/e2e-image/e2e-runner.sh]: This is unnecessary. kubetest already accepts `"${@}"`. Use the flag directly

File: scenarios/kubernetes_kops_aws.py [hunk 5]
```diff
@@ -165,6 +165,8 @@ def main(args):
                (args.cluster, zones, args.state, args.nodes, aws_ssh))
     if args.image:
         e2e_opt += ' --kops-image=%s' % args.image
+    if args.kops_args:
+        e2e_opt += ' --kops-args=\"%s\"' % args.kops_args
```
Comment (fejta) [inline scenarios/kubernetes_kops_aws.py]: Please use the flag directly (`cmd.append('--kops-args=K')`), do not rely on the magic E2E_OPT environment variable.

**Comment by fejta** [comment]
Thanks! However I want to do this without manipulating any environment variables for this.

File: scenarios/kubernetes_kops_aws.py (hunk 5 above)
Comment (wongma7) [inline scenarios/kubernetes_kops_aws.py]: `cmd` here is "docker run", not kubetest, though, so if I do that docker thinks I'm passing in a repo/tag

I want to believe this is why the other kubetest args (--kops-cluster) are forced to be passed in via E2E_OPT, we have to pipe it all the way through the python script -> docker run -> e2e-runner.sh -> kubetest.

Comment (wongma7) [inline scenarios/kubernetes_kops_aws.py]: actually you're right, it should work, --entrypoint is supposed to be exec form which lets us passs arguments so idk why docker is complaining. Let me try again...

Comment (fejta) [inline scenarios/kubernetes_kops_aws.py]: I suspect the person who added the previous flags was simply unaware of the ability to these as cmd line arguments

**Comment by wongma7** [comment]
@fejta PTAL, I ditched E2E_OPT. part of the problem was kops-e2e-runner.sh wasn't passing its args to e2e-runner.sh, fixed now

**Comment by fejta** [comment]
Fantabulous!
/lgtm
Do you need me to merge this for you?

File: scenarios/kubernetes_kops_aws.py [hunk 6]
```diff
@@ -198,6 +198,9 @@ def main(args):
 
     cmd.append(kubekins(args.tag))
 
+    if args.kops_args:
```
Comment (fejta) [inline scenarios/kubernetes_kops_aws.py]: This is pretty silly to fail line.

Feel free to set L68 to `too-many-locals,too-many-branches`

Comment (wongma7) [inline scenarios/kubernetes_kops_aws.py]: done, thanks :)

**Comment by fejta** [comment]
/assign @spxtr

Any idea why this test status is hung?

**Comment by spxtr** [comment]
@k8s-bot ok to test

**Comment by spxtr** [comment]
LGTM should count as ok to test...

**Comment by fejta** [comment]
Oh I missed that
